### ISSUE TITLE
_unique_card aggregation fails because of Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'decksite.d.person_id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by

### ISSUE THREAD

**Issue by vorpal-buildbot** [PennyDreadfulMTG/Penny-Dreadful-Tools#6931]
Reported on Discord by bakert#2193

**Comment by bakert** [comment]
We know we're getting a unique person_id because of the HAVING clause but I would hazard we can rewrite this in slightly saner SQL that will make mariadb less unhappy. This is blocking a PR.

```
SELECT
            card, person_id
        FROM
            deck_card AS dc
        INNER JOIN
            deck AS d ON dc.deck_id = d.id
        WHERE
            d.id IN (SELECT deck_id FROM deck_match GROUP BY deck_id HAVING COUNT(*) >= 3)
        GROUP BY
            card
        HAVING
            COUNT(DISTINCT person_id) = 1
```

fails during jenkins tests

**Comment by bakert** [comment]
https://travis-ci.org/PennyDreadfulMTG/Penny-Dreadful-Tools/jobs/622130187 is the log of the failure. I'm not sure why prod is ok but travis is not. I'd expect them to use the same db/setup.

I posted about it on Stack Overflow here: https://stackoverflow.com/questions/59758744/how-to-tell-mariadb-mysql-that-a-column-is-unique-due-to-having-clause

**Comment by bakert** [comment]
I rewrote the query to use a subquery to pass the check which revealed similar issues with the (more complex) trailblazer query.

I rewrote the trailblazer query to use a subquery instead of trying to fetch the id at the same time as doing the GROUP BY and it actually revealed that the original query was fetching out the wrong id. This is pretty huge and a good argument for having ONLY_FULL_GROUP_BY turned on everywhere.
